### DISCUSSION TITLE
Compute norm of a material gradient

### DISCUSSION THREAD

**Discussion by josebastiase** [idaholab/moose#27716] [Q&A Modules: Porous Flow]
Hi all,

I am studying the impact of material properties on my simulation results. Specifically, I want to evaluate how variations in the material properties affect the outcomes. To achieve this, I plan to use the norm of the gradient of the material property across the simulation space as a proxy measure.

Mathematically, I am considering the expression:

∥∇E∥2

where E represents the Young's modulus, which is assigned to the elements and stored in an auxiliary variable (AuxVar). The goal is to compute this gradient norm within my simulation to analyze the effect of spatial variations in Young's modulus.

Could anyone suggest the easiest way to compute this within my simulation?

Thanks in advance,
Jose

**Comment by GiudGiud** [comment]
Hello

If you want the average or the max of the norm of the gradient of the material property, then you want to create a new postprocessor

If you want to look at a field representing this quantity locally then use an auxiliary kernel. 

There may already be a pre implemented version of this. It's unlikely though given that there are two operations applied. If you don't want to write new code you could see if each operation can be applied separately to different auxiliary variables, in a chain

Guillaume

**Reply by josebastiase** [reply]
Thanks. Yes, I want to compute the field, so in my case, I would need to work with Auxiliary variables and kernels.

If I understood correctly I need to assign one aux kernel for each derivative in x,y,z and a final one to compute the gradient.

Is there a way to directly compute the gradient in one direction with an auxiliary kernel?

**Reply by GiudGiud** [reply]
I think there is a gradientAux 

Maybe functorGradientAux that would work
